src/plugins/components/PluginsDetailsPage/PluginsDetailsPage.tsx

```diff
@@ -0,0 +1,143 @@
+import Typography from "@material-ui/core/Typography";
+import AppHeader from "@saleor/components/AppHeader";
+import { ConfirmButtonTransitionState } from "@saleor/components/ConfirmButton";
+import Container from "@saleor/components/Container";
+import Form from "@saleor/components/Form";
+import Grid from "@saleor/components/Grid";
+import PageHeader from "@saleor/components/PageHeader";
+import SaveButtonBar from "@saleor/components/SaveButtonBar";
+import { sectionNames } from "@saleor/intl";
+import { maybe } from "@saleor/misc";
+import { UserError } from "@saleor/types";
+import { ConfigurationItemInput } from "@saleor/types/globalTypes";
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { Plugin_plugin } from "../../types/Plugin";
+import PluginInfo from "../PluginInfo";
+import PluginSettings from "../PluginSettings";
+
+export interface FormData {
+  active: boolean;
+  configuration: ConfigurationItemInput[];
+}
+
+export interface PluginsDetailsPageProps {
+  disabled: boolean;
+  errors: UserError[];
+  plugin: Plugin_plugin;
+  saveButtonBarState: ConfirmButtonTransitionState;
+  onBack: () => void;
+  onSubmit: (data: FormData) => void;
+}
+
+const PluginsDetailsPage: React.StatelessComponent<PluginsDetailsPageProps> = ({
+  disabled,
+  errors,
+  plugin,
+  saveButtonBarState,
+  onBack,
+  onSubmit
+}) => {
+  const intl = useIntl();
+  const initialForm: FormData = {
+    active: maybe(() => plugin.active, false),
+    configuration: maybe(() => plugin.configuration, [])
+  };
+
+  return (
+    <Form errors={errors} initial={initialForm} onSubmit={onSubmit}>
+      {({ data, errors, hasChanged, submit, set, triggerChange }) => {
+        const onChange = event => {
+          const newData = {
+            active: data.active,
+            configuration: data.configuration
+          };
+          const { name, value } = event.target;
+          name === "active"
+            ? (newData.active = value)
+            : (newData.active = data.active);
+          if (newData.configuration) {
+            newData.configuration.map(item => {
+              if (item.name === name) {
+                item.value = value;
+              }
+            });
+          }
+          triggerChange();
+          set(newData);
+        };
+        return (
+          <Container>
+            <AppHeader onBack={onBack}>
+              {intl.formatMessage(sectionNames.plugins)}
+            </AppHeader>
+            <PageHeader
+              title={`${maybe(() => plugin.name, "")} ${intl.formatMessage({
```

Comment:
```
intl.formatMessage({
  defaultMessage: "{pluginName} Details",
  description: "header"
}, {
  pluginName: maybe(() => plugin.name, '...')
})
```

We should translate whole sentences rather than naively joining pieces together.